lib/core/src/Cardano/Wallet.hs

```diff
@@ -573,11 +575,11 @@ restoreBlocks ctx wid blocks nodeTip = db & \DBLayer{..} -> do
     let bp = blockchainParameters cp
     let (filteredBlocks, cps) = NE.unzip $ applyBlocks @s blocks cp
     let slotPoolDelegations =
-            [ (slotId, poolId)
+            [ (slotId, cert)
             | let slots = view #slotId . view #header <$> blocks
```

Comment:
is it correct? we are taking `slots` from all `blocks`, but `delegations` from  `filteredBlocks` and then zip them :thinking: shouldn't we have `blocks` also for `delegations`. Otherwise we can have wrong slotId for some discovered certs?

Reply:
I also had the same concern on my first review. Looking closer, the wallet primitive does not change the order bor the number of elements of the prefiltered blocks so it should be fine. Yet, It's a bit error prone I agree.

I'll see if we can rewrite this a bit in a subsequent PR.

Reply:
yes, it is not intuitive. I was thinking about situation when we have in `block` for example `[slot1, slot2,slot3,slot4]` after viewing, but in `prefilteredBlock`  `[(slot3, cert3)] ----> [cert3]` and after zip we end up `[(slot1, cert3)]` :thinking:

Reply:
`prefilterBlock` does not discard blocks! It prefilters transactions **within** blocks. And returns a corresponding checkpoint for each block.

So, if you prefilter `[blk1, blk2, blk3]`, you get back `[cp1, cp2, cp3]` where checkpoints here contains only the information relevant to you. Yet, `cp1`, `cp2` and `cp3` may be roughly equal if there was no information relevant for your wallet.

Reply:
thanks for the explanation, then it should be good